plugins/slack_webhooks/test/samson_slack_webhooks/slack_webhooks_service_test.rb

```diff
@@ -0,0 +1,58 @@
+require_relative '../test_helper'
+
+SingleCov.covered!
+
+describe SamsonSlackWebhooks::SlackWebhooksService do
+  let(:deploy) { deploys(:succeeded_test) }
+  let(:service) { SamsonSlackWebhooks::SlackWebhooksService.new }
+
+  before do
+    Rails.cache.delete(:slack_users)
+  end
+
+  describe '#users' do
+    with_env(SLACK_API_TOKEN: 'some-token')
+
+    it "shows all users" do
+      stub_request(:post, "https://slack.com/api/users.list").
+        to_return(body: {ok: true, members: [{id: 1, name: 'NICK', profile: {image_48: 'AVATAR'}}]}.to_json)
+      service.users.must_equal([{id: 1, name: "NICK", avatar: "AVATAR", type: "contact"}])
+
+      # is cached
+      service.users.object_id.must_equal service.users.object_id
+    end
+
+    it "returns empty users when there's an error" do
+      stub_request(:post, "https://slack.com/api/users.list").
+        to_return(body: {ok: false, error: "some error"}.to_json)
+      Rails.logger.expects(:error).with('Error fetching slack users: some error')
+      service.users.must_equal([])
+
+      # is cached
+      service.users.object_id.must_equal service.users.object_id
```

Comment:
this is an interesting way to test! I like the idea, but I think it might be a little misleading if it is only really valid for the cache used in the test environment. Probably not worth changing, another way to express this might be to assert that the `Faraday` method isn't invoked.